kernel/CMakeLists.txt

```diff
@@ -104,6 +104,7 @@ target_sources_ifdef(
 target_include_directories(kernel PRIVATE
   ${ZEPHYR_BASE}/kernel/include
   ${ARCH_DIR}/${ARCH}/include
+  ${ZEPHYR_BASE}/subsys/pm
```

Comment:
Aesthetics, but this is breaking the component layering idea.  Subsystem code depends on kernel APIs, it should never be the reverse.  Maybe split out the low level from high level PM APIs and more them to the kernel?  Or just move all of pm into a kernel subdirectory, it's always going to be tied to the kernel anyway.

Reply:
Yep, I got this felling when including this directory. Moving the whole subsystem to inside the kernel is something that makes sense for me since they are tied together. My push back is doing it after feature freezing. The problem this pr address is real, so we need the fix. One solution for now would be move this API and the cpu initialized counter to the kernel and make the pm subsystem call it instead. What do you think ?

Reply:
To be fair, this dependency always existed, `kernel/idle.c` is "depending" on pm subsystem. The difference here is that the include path is explicitly needed. This notification should come from kernel since it is an async event. Ideas about how to do it without moving pm subsystem to kernel now ? I don't want to rush it since it is not only this part but also device power management that would possibly move.